sdks/java/io/hcatalog/src/main/java/org/apache/beam/sdk/io/hcatalog/HCatalogIO.java

```diff
@@ -123,12 +129,147 @@ public static Read read() {
     return new AutoValue_HCatalogIO_Read.Builder().setDatabase(DEFAULT_DATABASE).build();
   }
 
+  /**
+   * Read unbounded data from Hive. This essentially means that as new partitions show up, we start
+   * reading data for that partition.
+   */
+  public static UnboundedRead unbounded() {
```

Comment:
Are you sure you need a new class for this, can you just add a boolean parameter to Read "isUnbounded" and add an if() statement in expand?

Reply:
That was probably an ill advise of my part (there was  a totally different approach, not the read based one too). @jhalaria I think this is probablysimilar in principle to what we called the unification, maybe worth to tackle it for this PR.